http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -965,6 +974,62 @@ class GKEInterpreter[F[_]](
 
     } yield ()
 
+  private[util] def installCromwellApp(helmAuthContext: AuthContext,
+                                       appName: AppName,
+                                       release: Release,
+                                       cluster: KubernetesCluster,
+                                       nodepoolName: NodepoolName,
+                                       namespaceName: NamespaceName)(implicit ev: Ask[F, AppContext]): F[Unit] = {
+    val chart = config.cromwellAppConfig.chart
+
+    for {
+      ctx <- ev.ask
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Installing helm chart ${chart} for app ${appName.value} in cluster ${cluster.getGkeClusterId.toString}"
+      )
+
+      chartValues = buildCromwellAppChartOverrideValuesString(appName, cluster, nodepoolName, namespaceName)
+      _ <- logger.info(ctx.loggingCtx)(s"Chart override values are: $chartValues")
+
+      // Invoke helm
+      helmInstall = helmClient
+        .installChart(
+          release,
+          chart.name,
+          chart.version,
+          org.broadinstitute.dsp.Values(chartValues.mkString(",")),
+          false
+        )
+        .run(helmAuthContext)
+
+      // Currently we always retry.
+      // The main failure mode here is helm install, which does not have easily interpretable error codes
+      retryConfig = RetryPredicates.retryAllConfig
+      _ <- tracedRetryF(retryConfig)(
+        helmInstall,
+        s"helm install for app ${appName.value} in project ${cluster.googleProject.value}"
+      ).compile.lastOrError
+
+      // Poll the app until it starts up
+      last <- streamFUntilDone(
```

Comment:
you could use [streamUntilDoneOrTimeout](https://github.com/broadinstitute/workbench-libs/blob/develop/google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/package.scala#L164) instead, and then `adaptError` to proper error